src/app/modules/notification/services/notification.service.ts

```diff
@@ -0,0 +1,111 @@
+import { Injectable } from '@angular/core';
+
+import { Notification } from '../@type/notification';
+import { NotificationOptions } from '../@type/notification-options';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class NotificationService {
+
+  /** Default notifiactions options */
+  options: NotificationOptions = {
+    isDismissable: true,
+    isSelfClosing: true,
+    duration: 5
+  };
+
+  /** Notifications list */
+  private notifications: Notification[] = [];
+
+  /** Id prefix of the nofications */
+  private idPrefix = 'notification-';
+
+  /** Delay to wait before removing a notification on a dismiss action */
+  private removingDelay = 1000;
+
+  constructor() { }
+
+  /**
+   * Get all active notifications
+   */
+  getNotifications(): Notification[] {
+    return this.notifications;
+  }
+
+  /**
+   * Get removing delay
+   */
+  getRemovingDelay(): number {
+    return this.removingDelay;
+  }
+
+  /**
+   * Add a notification
+   */
+  push(message: string, type: string = 'default', options = {}): void {
+    const notification = {
+      id: this.generateUniqueId(),
+      message: message,
+      type: type,
+      options: this.mergeWithDefaultOptions(options)
+    };
```

Comment:
You can just write:
```
      id: this.generateUniqueId(),
      message,
      type,
      options: this.mergeWithDefaultOptions(options)
```
If the key and value are the same you can just put the key ;)

Reply:
Good tip :wink: